src/core/lib/math/distributiongenerator.cpp

```diff
@@ -35,12 +35,142 @@
+            errMsg += dlsym_error;
+            dlclose(libraryHandle);
+            OPENFHE_THROW(errMsg);
+        }
+        // std::cerr << "InitPRNGEngine: using external PRNG" << std::endl;
+    }
+} 
+
+PRNG& PseudoRandomNumberGenerator::GetPRNG() {
+    // initialization of PRNGs
+    if (m_prng == nullptr) {
+#pragma omp critical
+        {
+            if (!genPRNGEngine)
+                InitPRNGEngine();
+
+            std::array<uint32_t, PRNG::MAX_SEED_GENS> seed{};
+  #if defined(FIXED_SEED)
+            // Only used for debugging in the single-threaded mode.
+            std::cerr << "**FOR DEBUGGING ONLY!!!!  Using fixed initializer for PRNG. "
+                         "Use a single thread only, e.g., OMP_NUM_THREADS=1!"
+                      << std::endl;
+
+            seed[0] = 1;
+  #else
+            // A 512-bit seed is generated for each thread (this roughly corresponds
+            // to 256 bits of security). The seed is the sum of a random sample
+            // generated using std::random_device (typically works correctly in
+            // Linux, MacOS X, and MinGW starting with GCC 9.2) and a PRNG sample
+            // seeded from current time stamp, a hash of the current thread, and a
+            // memory location of a heap variable. The PRNG sample is added in
+            // case random_device is deterministic (happens on MinGW with GCC
+            // below 9.2). All future calls to PRNG use the seed generated here.
+
+            // The code below derives randomness from time, thread id, and a memory
+            // location of a heap variable. This seed is relevant only if the
+            // implementation of random_device is deterministic (as in older
+            // versions of GCC in MinGW)
+            std::array<uint32_t, PRNG::MAX_SEED_GENS> initKey{};
+            // high-resolution clock typically has a nanosecond tick period
+            // Arguably this may give up to 32 bits of entropy as the clock gets
+            // recycled every 4.3 seconds
+            initKey[0] = std::chrono::high_resolution_clock::now().time_since_epoch().count();
+            // A thread id is often close to being random (on most systems)
+            initKey[1] = std::hash<std::thread::id>{}(std::this_thread::get_id());
+            // On a 64-bit machine, the thread id is 64 bits long
+            // skip on 32-bit arm architectures
+    #if !defined(__arm__) && !defined(__EMSCRIPTEN__)
+            if (sizeof(size_t) == 8)
+                initKey[2] = (std::hash<std::thread::id>{}(std::this_thread::get_id()) >> 32);
+    #endif
+            // heap variable; we are going to use the least 32 bits of its memory
+            // location as the counter. This will increase the entropy of the PRNG sample
+            void* mem        = malloc(1);
+            uint32_t counter = reinterpret_cast<long long>(mem);  // NOLINT
+            free(mem);
+
+            std::uniform_int_distribution<uint32_t> distribution(0);
```

Comment:
Does std::uniform_int_distribution actually do anything here?  We expect the output of the PRNG to already be uniformly-distributed.  This has the appearance of a complicating no-op.  AmIRight?